site-builder/src/site/resource.rs

```diff
@@ -458,3 +487,118 @@ pub(crate) fn full_path_to_resource_path(full_path: &Path, root: &Path) -> Resul
             .ok_or(anyhow!("could not process the path string: {:?}", rel_path))?
     ))
 }
+
+#[cfg(test)]
+mod tests {
+    use std::{collections::BTreeMap, path::PathBuf};
+
+    use super::{HttpHeaders, ResourceManager};
+    use crate::{site::config::WSResources, walrus::Walrus};
+    struct PatternMatchTestCase {
+        pattern: &'static str,
+        path: &'static str,
+        expected: bool,
+    }
+
+    #[test]
+    fn test_is_pattern_match() {
+        let tests = vec![
+            PatternMatchTestCase {
+                pattern: "*.txt",
+                path: "file.txt",
```

Comment:
Question: shouldn't paths and patterns be all in the form `/some/pattern/*` and `/some/path` with a leading `/`?

Reply:
Yes makes sense. Updated the tests. ✅